src/tests/net.rs

```diff
@@ -6,68 +6,92 @@ use rand::{
 };
 
 use crate::{
-    alkahest, read_packet, write_packet_to_vec, Formula, Lazy, SerIter, Serialize, SerializeRef,
+    alkahest, read_packet, write_packet_to_vec, Deserialize, Formula, Lazy, SerIter, Serialize, SerializeRef, Ref,
 };
 
+#[alkahest(Formula)]
+pub enum GameMessageFormula {
+    Client(Ref<ClientMessageFormula>),
```

Comment:
The `Ref` looks unnecessary here

Reply:
Ah yeah, I was just experimenting with it -- this is actually the reason I initially wanted to separate the formula from the serialize and deserialize types, to insert a wrapping handler in the mix. Ultimately I'm trying to play with making a `Ref`-like wrapper that would help to do schema evolution (forward/backward compatibility). But it is of course unnecessary in this case.

Reply:
What exactly are properties of `Ref`-like wrapper you have in mind?
Current `Ref` IMHO is enough to allow separate evolution of field's `Formula`.
Currently user may evolve schema and manually ensure desired compatibility.

I have the following plan to better support them:
* Add validation to check compatibility between two `Formula`s.
* Add attributes to mark new fields when deriving `Formula`. Then compatibility will be checked between versions. All in proc-macro execution time.

Reply:
Hm, not sure if I quite follow. If you serialize some data with `Ref<SomeTV1>`, add a field, then try to deserialize the old data with `Ref<SomeTV2>`, that will not work, right?

It would be nice to have the ability to provide default values for fields added and then based on size of serialized data detect which fields are missing and fill them in.

You could also put another wrapper that is like `Lazy<Ref<T>>` where instead of filling in with default values, you could "attempt to get" the type at different versions, i.e. with different numbers of fields. Similar to "probe" types I made in `protoss` experiment for `rkyv`: https://github.com/fu5ha/protoss/blob/master/protoss_test/src/derive.rs#L169-L173

Reply:
Yes, you would need to have defaults for new fields for backward compatibility.
For forward compatibility it would be enough to mark formula as non-exhaustive, so it would just skip fields added at the end.
How about `Lazy<AnyRef>` that can be viewed as `Lazy<AnyRef<T>>`.
Where the only difference between `Ref<T>` and `AnyRef<T>` is that binary format of `AnyRef` doesn't depend on `T`

Reply:
Maybe `Lazy` layer is not needed.
```rust

struct AnyRef {...}

impl AnyRef {
  fn decode<F: Formula, T: Deserialize<F>>() -> Option<T> { ... }
}
```

This requires more work 🤣

Reply:
Yeah exactly, I was thinking of making a single thing similar to what you propose that's like a hybrid.

Ideally it would still be typed based on the main type it represents "some version" of. Then `decode` would only let you ask for one of the known versions of that type

Reply:
```rust
struct AnyVersionRef<T: Versioned> {...}

impl<T: Versioned> AnyVersionRef<T> {
  fn decode_version<V: VersionOf<T>>(&self) -> Option<V> { ... }
  fn latest_with_defaults(self) -> T { ... }
}
```

Needs more thinking about how exactly to integrate the deserialize and formula types/bounds but thats the basic vision I guess.